catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsTest.java

```diff
@@ -0,0 +1,500 @@
+        when(resourceResponseMock.getResource()).thenReturn(resourceMock);
+
+ 
+        frameworkProperties.setReliableResourceDownloadManager(reliableResourceDownloadManagerMock);
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+    }
+
+    @Test(expected = DataUsageLimitExceededException.class)
+    public void testGetResourceCatchesDataUsageLimitExceededException()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException {
+
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getProperties()).thenThrow(DataUsageLimitExceededException.class);
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+        verify(resourceRequestMock).getProperties();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceCatchesRuntimeException()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException {
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getProperties()).thenThrow(RuntimeException.class);
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+        verify(resourceRequestMock).getProperties();
+    }
+
+    @Test(expected = ResourceNotSupportedException.class)
+    public void testGetResourceCatchesStopProcessingException()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException {
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getProperties()).thenThrow(StopProcessingException.class);
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+        verify(resourceRequestMock).getProperties();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceNullSourceNameNotEnterpriseThrowsResourceNotFoundException()
+            throws Exception {
+
+        boolean isEnterprise = false;
```

Comment:
Could make these three variables class variables and set them in your setup method. Then just changed them as needed for the test. Looks like most of them wouldn't need to make changes at all.